tdrs-frontend/src/components/FileUpload/utils.jsx

```diff
@@ -140,3 +140,45 @@ export const tryGetUTF8EncodedFile = async function (fileBytes, file) {
     return file
   }
 }
+
+// import SET_SELECTED_YEAR
+// import SET_SELECTED_QUARTER
+
+export const checkHeaderFile = async function (
+  result,
+  file,
+  fiscalSelectedYear,
+  fiscalSelectedQuarter
+) {
+  const CalendarToFiscalYearQuarter = (calendarYear, calendarQuarter) => {
+    let quarter = parseInt(calendarQuarter)
+    let year = parseInt(calendarYear)
+    if (quarter === 4) {
+      year = year + 1
+    }
+    quarter = quarter === 4 ? 1 : quarter + 1
+    return [year.toString(), quarter.toString()]
+  }
+  // reads the first line of file formatted as UTF8Encoded
+  const decoder = new TextDecoder('utf-8') // Or another encoding if needed
```

Comment:
@raftmsohani I'm noticing an edge case where the wrong quarter is allowed to be submitted (but is rejected). its related to the encoding. test file below, which is UTF16LE encoded. 

[2024.Q1.Active Case Data_UTF16LE.txt](https://github.com/user-attachments/files/19455724/2024.Q1.Active.Case.Data_UTF16LE.txt)

Reply:
sure. could you spin up the ticket so I can approve this with links to followup @raftmsohani ?